### DISCUSSION TITLE
Support OAS 3.1

### DISCUSSION THREAD

**Discussion by black-snow** [Kong/insomnia#3577] [Ideas]
**Is your feature request related to a problem? Please describe.**
OAS 3.1 hasn't settled yet but is quite close to being final (afaik). I couldn't find any issue / roadmap tracking progress for this - please close the issue & hint me if I'm blind.

**Describe the solution you'd like**
Support [OAS 3.1 RC0](https://www.openapis.org/blog/2020/06/18/openapi-3-1-0-rc0-its-here) / JSON-Schema 2019-09 and keep on improving until it's final.

**Comment by nijikokun** [comment]
Good call-out @black-snow still waiting for the third-party libraries to catch up 👀

**Comment by black-snow** [comment]
not stale

Haven't check for a while, has anything happened here in the meantime @nijikokun ? https://spec.openapis.org/oas/v3.1.0 is out.

**Reply by black-snow** [reply]
Am I the only one working with 3.1+? :D

**Reply by Stokestack** [reply]
No. I just tried to open my OpenAPI file and Insomnia complained because it specifies version 3.1.

**Comment by dimitropoulos** [comment]
Could you give a specific example, please, of something you need from OAS 3.1?  It will help get the ball rolling.

**Reply by black-snow** [reply]
sadly, it's now been a while since I had to doc an API so I can't be super specific - I remember coming from JSON Schema, example vs. examples was a small issue -- I don't fully remember, it's been almost a year :/

**Reply by tfedor** [reply]
@dimitropoulos e.g. array type, multiple examples or updated JSON Schema support to 2020-12? Pretty please

**Comment by Stokestack** [comment]
This is a real issue. I don't remember all of what's added in 3.1, but it's very important to be able to put descriptions alongside the use of references. I can't imagine why this was ever prohibited.

For example, if you've defined a schema called Rectangle and you want to use it by reference in various places in your API, pre-3.1 OpenAPI wouldn't let you describe what you were using it for in each instance. Baffling.

**Comment by dimitropoulos** [comment]
for what it's worth: our upstream already merged support for OSA 3.1 https://github.com/APIDevTools/swagger-parser/pull/191 but they're having trouble making an npm release: https://github.com/APIDevTools/swagger-parser/pull/191

**Reply by MrStanDu33** [reply]
So we're stuck atm with OA3.0 ? 😒

**Reply by black-snow** [reply]
for now

**Reply by miketheman** [reply]
Looks like they resolved the release issue 6 days ago 🎉

**Reply by brookmg** [reply]
Is this issue resolved?

**Reply by black-snow** [reply]
haven't checked for a while - might be able to on monday

**Reply by Wazbat** [reply]
Looking forward to this. There's a few nifty features in 3.1 I'd love to use
